packages/docs/src/components/navigator.tsx

```diff
@@ -185,9 +209,14 @@ export class Navigator extends React.PureComponent<INavigatorProps, INavigatorSt
     }
 
     private selectNext(direction = 1) {
-        return () => this.setState({
-            ...this.state,
-            selectedIndex: Math.max(0, this.state.selectedIndex + direction),
-        });
+        return () => {
+            // indicate that the selected item may need to be scrolled into view after update.
+            // this is not possible with mouse hover cuz you can't hover on something off screen.
+            this.shouldCheckSelectedInViewport = true;
```

Comment:
my first attempt was a `ref` handler that only got attached if `isSelected`:

```
ref={isSelected ? this.maybeScrollIntoView : undefined}
```

and it worked beautifully, except it also happened during mouse hover. this would cause the scroll position to freak out as the contents scrolled under the mouse, triggering more scrolling, etc.

i found this flag + `componentDidUpdate` to be an effective solution yet i'm open to other pattern suggestions. (using a flag instead of state because i do _not_ want changes to this property to cause re-renders: it only affects the post-render behavior.)